lib/tasks/server/express-server.js

```diff
@@ -153,29 +153,30 @@ class ExpressServerTask extends Task {
     this.scheduleServerRestart();
   }
 
-  restartHttpServer() {
+  async restartHttpServer() {
     if (!this.serverRestartPromise) {
-      this.serverRestartPromise = this.stopHttpServer()
-        .then(() => {
-          this.invalidateCache(this.serverRoot);
-          return this.startHttpServer();
-        })
-        .then(() => {
-          this.emit('restart');
-          this.ui.writeLine('');
-          this.ui.writeLine(chalk.green('Server restarted.'));
-          this.ui.writeLine('');
-        })
-        .catch((err) => {
-          this.ui.writeError(err);
-        })
-        .finally(() => {
-          this.serverRestartPromise = null;
-        });
+      try {
+        this.serverRestartPromise = await this.stopHttpServer()
```

Comment:
I'm not sure about the `await` here. `serverRestartPromise` will no longer contain the promise in this case, but instead the resolution value of the promise 🤔

Reply:
Good point! I think with the new commit the correct logic is preserved.